src/grt/src/fastroute/src/FastRoute.cpp

```diff
@@ -520,33 +519,35 @@ void FastRouteCore::setLayerOrientation(int x)
 
 void FastRouteCore::addPin(int netID, int x, int y, int layer)
 {
-  nets[netID]->pinX.push_back(x);
-  nets[netID]->pinY.push_back(y);
-  nets[netID]->pinL.push_back(layer);
+  FrNet* net = nets[netID];
+  net->pinX.push_back(x);
+  net->pinY.push_back(y);
+  net->pinL.push_back(layer);
 }
 
 int FastRouteCore::addNet(odb::dbNet* db_net,
-                          int nPins,
-                          int validPins,
+                          int num_pins,
                           float alpha,
-                          bool isClock,
+                          bool is_clock,
+                          int driver_idx,
                           int cost,
-                          std::vector<int> edgeCostPerLayer)
+                          std::vector<int> edge_cost_per_layer)
 {
   int netID = newnetID;
-  pinInd = validPins;
+  pinInd = num_pins;
   MD = std::max(MD, pinInd);
   nets[newnetID]->db_net = db_net;
```

Comment:
time to stop copying the clueless code that repeats the array reference.
this should factor nets[newnetID] to a local variable